### PR TITLE
correct handling of xmlhttprequest network errors

### PR DESCRIPTION
**Pull request by sirpy**
Currently connection errors are uncaught. For example try to close your wifi.

## Description
add .onerror handler to reject the promise in case of an xmlhttprequest network error

## Type of change

- [x] Bug fix 

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no warnings.
- [x] I have updated or added types for all modules I've changed
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run test``` in the root folder with success and extended the tests if necessary.
- [x] I ran ```npm run build``` in the root folder and tested it in the browser and with node.
- [x] I ran ```npm run dtslint``` in the root folder and tested that all my types are correct
- [x] I have tested my code on an ethereum test network.

### CODE REVIEW

File: packages/web3-providers/src/providers/HttpProvider.js [hunk 1]
```diff
@@ -180,7 +180,11 @@ export default class HttpProvider {
 
                 reject(new Error(`Connection error: Timeout exceeded after ${this.timeout}ms`));
             };
-
+            
+            request.onerror = () => {
+                this.connected = false;
+                reject(new Error(`Network error`))
+            }
```
Comment (nivida) [inline packages/web3-providers/src/providers/HttpProvider.js]: could you add the missing break, update the error message (adding of payload), and could you also add the missing semicolon on line 186. The test should get extended as well.

**Review by nivida — CHANGES_REQUESTED** [review]

File: packages/web3-providers/src/providers/HttpProvider.js [hunk 2]
```diff
@@ -180,7 +180,11 @@ export default class HttpProvider {
 
                 reject(new Error(`Connection error: Timeout exceeded after ${this.timeout}ms`));
             };
-
+            
+            request.onerror = () => {
+                this.connected = false;
+                reject(new Error(`Network error`))
```
Comment (sirpy) [inline packages/web3-providers/src/providers/HttpProvider.js]:
```suggestion
                reject(new Error(`Network error`));
```

File: packages/web3-providers/src/providers/HttpProvider.js [hunk 3]
```diff
@@ -180,7 +180,11 @@ export default class HttpProvider {
 
                 reject(new Error(`Connection error: Timeout exceeded after ${this.timeout}ms`));
             };
-
+            
+            request.onerror = () => {
+                this.connected = false;
```
Comment (sirpy) [inline packages/web3-providers/src/providers/HttpProvider.js:185]:
```suggestion
                this.connected = false;
                
```

File: packages/web3-providers/src/providers/HttpProvider.js (hunk 2 above)
Comment (sirpy) [inline packages/web3-providers/src/providers/HttpProvider.js]:
```suggestion
                reject(new Error(`Network error ${JSON.stringify(payload)}`))
```

File: packages/web3-providers/src/providers/HttpProvider.js [hunk 4]
```diff
@@ -180,7 +180,12 @@ export default class HttpProvider {
...
+                
+                reject(new Error(`Network error`));
```
Comment (sirpy) [inline packages/web3-providers/src/providers/HttpProvider.js]:
```suggestion
                reject(new Error(`Network error ${JSON.stringify(payload)}`))
```

File: packages/web3-providers/src/providers/HttpProvider.js (hunk 1 above)
Comment (sirpy) [reply packages/web3-providers/src/providers/HttpProvider.js]: @nivida done

**Comment by sirpy** [comment]
@nivida can you review it again

**Comment by nivida** [comment]
The semicolon and the additional test case is still missing @sirpy :)

**Comment by sirpy** [comment]
> The semicolon and the additional test case is still missing @sirpy :)

add the changes you want using github suggest @nivida

**Pull request merged**